packages/trader/src/AppV2/Components/TradeParameters/Duration/day.tsx

```diff
@@ -221,16 +277,26 @@ const DayInput = ({
                     <ActionSheet.Footer
                         alignment='vertical'
                         shouldCloseOnPrimaryButtonClick={false}
+                        isPrimaryButtonDisabled={is_disabled}
                         primaryAction={{
                             content: <Localize i18n_default_text='Done' />,
                             onAction: () => {
-                                setOpen(false);
-                                setOpenTimePicker(false);
-                                if (formatted_date !== formatted_current_date) {
-                                    setEndTime('');
-                                }
-                                if (timeToMinutes(adjusted_start_time) > timeToMinutes(end_time)) {
-                                    setEndTime(adjusted_start_time);
+                                if (!is_disabled) {
+                                    setEndDate(temp_end_date);
+                                    setOpen(false);
+                                    setOpenTimePicker(false);
+                                    const end_date_temp = temp_end_date.toLocaleDateString('en-GB', {
```

Comment:
whats the flow / which one should be used when, given the:
end_date_temp, temp_end_date, end_date :D :D :D

Reply:
ah yeah the names got confusing , let me rename thanks

Reply:
So actually I added temp_end_date inside the action sheet as it saves the value temporarily inside the action sheet .
-> once we save the date , the value is moved to main duration component as end_date and temp_end_date is moved to default value.
-> if  action sheet is closed without saving . temp_end_date is also changed to default value.

I changed the name to `**end_date_input**` now as it was making more sense rather then temp-. 

Thanks